File: Exemplos SmartPOS/App_IntentDigitalHUB_Java/Elgin_IntentDigitalHUB_SmartPOS_Java-font/app/src/main/java/com/elgin/java_intentdigitalhub_smartpos/IntentDigitalHubService/TERMICA/Commands/ImpressaoQRCode.java
package com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.Commands;

import com.elgin.java_intentdigitalhub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand;
import com.google.gson.JsonObject;

public class ImpressaoQRCode extends TermicaCommand {
    final private String dados;
    final private int tamanho;
    final private int nivelCorrecao;

    public ImpressaoQRCode(String dados, int tamanho, int nivelCorrecao) {
        super("ImpressaoQRCode");
        this.dados = dados;
        this.tamanho = tamanho;
        this.nivelCorrecao = nivelCorrecao;
    }

    @Override
    protected JsonObject functionParametersJson() {
        JsonObject functionParametersJson = new JsonObject();

        functionParametersJson.addProperty("dados", this.dados);
        functionParametersJson.addProperty("tamanho", this.tamanho);
        functionParametersJson.addProperty("nivelCorrecao", this.nivelCorrecao);

        return functionParametersJson;
    }
}

File: Exemplos SmartPOS/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_SmartPOS_Kotlin-font/app/src/main/java/com/elgin/kotlin_intent_digital_hub_smartpos/IntentDigitalHubService/TERMICA/Commands/ImpressaoQRCode.kt
package com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.Commands

import com.elgin.kotlin_intent_digital_hub_smartpos.IntentDigitalHubService.TERMICA.TermicaCommand
import com.google.gson.JsonObject


open class ImpressaoQRCode(
    private val dados: String,
    private val tamanho: Int,
    private val nivelCorrecao: Int)
    : TermicaCommand("ImpressaoQRCode") {
    override fun functionParametersJson(): JsonObject {
        val functionParametersJson = JsonObject()
        functionParametersJson.addProperty("dados", dados)
        functionParametersJson.addProperty("tamanho", tamanho)
        functionParametersJson.addProperty("nivelCorrecao", nivelCorrecao)
        return functionParametersJson
    }
}
